### PR TITLE
Fix handling of --registries-conf

### PR DESCRIPTION
**Pull request by nalind**
Instead of ignoring the global --registries-conf option and using only $REGISTRIES_CONFIG_PATH, use it for the option default.

### CODE REVIEW

File: pull.go [hunk 1]
```diff
@@ -167,14 +167,14 @@ func pullImage(ctx context.Context, store storage.Store, imageName string, optio
 	}()
 
 	logrus.Debugf("copying %q to %q", spec, destName)
-	err = cp.Image(ctx, policyContext, destRef, srcRef, getCopyOptions(options.ReportWriter, options.SystemContext, nil, ""))
+	err = cp.Image(ctx, policyContext, destRef, srcRef, getCopyOptions(options.ReportWriter, sc, nil, ""))
```
Comment (TomSweeneyRedHat) [inline pull.go]: In new.go:newBuilder, we tease out the systemContext from options and then pass it down to here without using it prior that I can see.  Could we just drop systemContext as a param to several functions and just pull the value out from options here in this code?  No a change necessary for this PR, but maybe a touchup issue for later.

Comment (nalind) [reply pull.go]: We'd have to call `getSystemContext()` to override the default signature policy path with the configured one for each case, but that's not a deal-breaker.

Comment (TomSweeneyRedHat) [reply pull.go]: yeah, and signaturepolicy should be in the the options too.

**Comment by TomSweeneyRedHat** [comment]
bot, retest this please.

**Comment by nalind** [comment]
Basically everything needs #855 at this time.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**